src/api/routes/getMod.ts

```diff
@@ -157,24 +157,28 @@ export class GetModRoutes {
 
         this.app.get(`/api/hashlookup`, async (req, res) => {
             // #swagger.tags = ['Mods']
-            // #swagger.summary = 'Show a mod that has a file with the specified hash.'
-            // #swagger.description = 'Show a mod that has a file with the specified hash. This is useful for finding the mod that a file belongs to.'
-            // #swagger.responses[200] = { description: 'Returns the mod.' }
+            // #swagger.summary = 'Get a specific mod version that has a file with the specified hash.'
+            // #swagger.description = 'Get a specific mod version that has a file with the specified hash. This is useful for finding the mod that a file belongs to.'
+            // #swagger.responses[200] = { description: 'Returns the mod version.' }
             // #swagger.responses[400] = { description: 'Missing hash.' }
             // #swagger.responses[404] = { description: 'Hash not found.' }
             // #swagger.parameters['hash'] = { description: 'The hash to look up.', type: 'string', required: true }
-            let hash = Validator.zString.min(8).safeParse(req.query.hash);
+            // #swagger.parameters['raw'] = { description: 'Return the raw mod depedendcies without attempting to resolve them.', type: 'boolean' }
+
+            const hash = Validator.zString.min(8).safeParse(req.query.hash).data;
+            const raw = Validator.zBool.safeParse(req.query.raw).data;
+
             if (!hash) {
                 return res.status(400).send({ message: `Missing hash.` });
             }
 
-            for (let version of DatabaseHelper.cache.modVersions) {
-                if (version.zipHash === hash.data) {
-                    return res.status(200).send({ mod: version.modId });
+            for (const version of DatabaseHelper.cache.modVersions) {
+                if (version.zipHash === hash) {
+                    return res.status(200).send({ modVersion: await (raw ? version.toRawAPIResonse() : version.toAPIResonse()) });
                 }
-                for (let fileHash of version.contentHashes) {
-                    if (fileHash.hash === hash.data) {
-                        return res.status(200).send({ mod: version.modId });
+                for (const fileHash of version.contentHashes) {
+                    if (fileHash.hash === hash) {
+                        return res.status(200).send({ modVersion: await (raw ? version.toRawAPIResonse() : version.toAPIResonse()) });
```

Comment:
The only worry (and reason why I returned only returned the mod ID) is that mods can share file hashes between versions. For example, with the original SongLoader mod, the map One More Time was included & afaik did not change between versions, whereas the actual plugin DLL file did.

It could lead to a situation where the modVersion that gets returned is a different version from whats installed. If people don't have an issue with this then I have no problem with it.

CC: @moddingpink since you initially asked for this

Reply:
@Saeraphinx 

Ah yeah, I see. For mod installers, this shouldn’t be an issue since we only check the "Plugins" and "Libs" folders anyway

Alternatively, to provide a more accurate response, we could return a list of mod versions that use this hash :thinking:

But ultimately, these changes i've made don't make a difference, as we still have the mod ID in the mod version that's returned

Reply:
Agreed. I'll return an array of modverstions that contain the hash.